tests/foreman/cli/test_computeresource_osp.py

```diff
@@ -63,47 +70,51 @@ def test_crud_and_duplicate_name(self, request, id_type):
             {
                 'name': name,
                 'provider': 'Openstack',
-                'user': OSP_SETTINGS['username'],
-                'password': OSP_SETTINGS['password'],
-                'tenant': OSP_SETTINGS['tenant'],
-                'url': OSP_SETTINGS['hostname'],
-                'project-domain-id': OSP_SETTINGS['project_domain_id'],
+                'user': OSP_SETTINGS.username,
+                'password': OSP_SETTINGS.password,
+                'tenant': OSP_SETTINGS.tenant,
+                'project-domain-id': OSP_SETTINGS.project_domain_id,
+                'url': osp_version,
             }
         )
         request.addfinalizer(lambda: self.cr_cleanup(compute_resource['id']))
         assert compute_resource['name'] == name
-        assert ComputeResource.exists(search=(id_type, compute_resource[id_type]))
+        assert target_sat.cli.ComputeResource.exists(search=(id_type, compute_resource[id_type]))
 
         # negative create with same name
-        with pytest.raises(CLIFactoryError):
-            make_compute_resource(
+        with pytest.raises(CLIReturnCodeError):
+            target_sat.cli.ComputeResource.create(
                 {
                     'name': name,
                     'provider': 'Openstack',
-                    'user': OSP_SETTINGS['username'],
-                    'password': OSP_SETTINGS['password'],
-                    'tenant': OSP_SETTINGS['tenant'],
-                    'url': OSP_SETTINGS['hostname'],
-                    'project-domain-id': OSP_SETTINGS['project_domain_id'],
+                    'user': OSP_SETTINGS.username,
+                    'password': OSP_SETTINGS.password,
+                    'tenant': OSP_SETTINGS.tenant,
+                    'project-domain-id': OSP_SETTINGS.project_domain_id,
+                    'url': osp_version,
                 }
             )
         # update
         new_name = gen_string('alpha')
-        ComputeResource.update({id_type: compute_resource[id_type], 'new-name': new_name})
+        target_sat.cli.ComputeResource.update(
+            {id_type: compute_resource[id_type], 'new-name': new_name}
+        )
         if id_type == 'name':
-            compute_resource = ComputeResource.info({'name': new_name})
+            compute_resource = target_sat.cli.ComputeResource.info({'name': new_name})
         else:
-            compute_resource = ComputeResource.info({'id': compute_resource['id']})
+            compute_resource = target_sat.cli.ComputeResource.info({'id': compute_resource['id']})
         assert new_name == compute_resource['name']
-        ComputeResource.delete({id_type: compute_resource[id_type]})
-        assert not ComputeResource.exists(search=(id_type, compute_resource[id_type]))
+        target_sat.cli.ComputeResource.delete({id_type: compute_resource[id_type]})
+        assert not target_sat.cli.ComputeResource.exists(
+            search=(id_type, compute_resource[id_type])
+        )
```

Comment:
There's also delete being run in finalizer in teardown, so maybe this delete() can be removed OR this can be moved to a new finalizer func OR remove the previous one?

Reply:
Updated.